packages/openactive-integration-tests/test/helpers/logger.js

```diff
@@ -105,35 +105,116 @@ class BaseLogger {
     this.recordLogEntry({
       type: "validations",
       stage: stage,
-      validations: data
+      validations: data,
     });
   }
 
-  async writeMeta() {
-    let data = _(this).omit([
-      'suite'
-    ]);
+  async writeMeta () {
+    let data = _.chain(this).omit([
+      "suite",
+    ]).value();
 
     let json = JSON.stringify(data, null, 4);
 
     await fs.writeFile(this.metaPath, json);
   }
 
-  get suiteName() {
-    throw Error('suiteName unimplemented');
+  setJestContext (context) {
+    this.jestContext = context;
+  }
+
+  get uniqueSuiteName () {
+    throw Error("suiteName unimplemented");
+  }
+
+  get metaPath () {
+    return `./output/json/${this.uniqueSuiteName}.json`;
+  }
+
+  get markdownPath () {
+    return `./output/${this.uniqueSuiteName}.md`;
+  }
+
+  get validationStatusCounts () {
+    if (this._validationStatusCounts) return this._validationStatusCounts;
+
+    let statuses = _.chain(this.logs)
+      .filter(log => log.type === "validations")
+      .flatMap(log => log.validations)
+      .countBy(log => log.severity)
+      .value();
+
+    return this._validationStatusCounts = {
+      ...statuses,
+      suggestion: statuses.suggestion || 0,
+      warning: statuses.warning || 0,
+      failure: statuses.failure || 0,
+    };
+  }
+
+  get specStatusCounts () {
+    if (this._specStatusCounts) return this._specStatusCounts;
+
+    let statuses = _.chain(this.logs)
+      .filter(log => log.type === "spec")
+      .filter(log => log.title !== "passes validation checks")
```

Comment:
Why would a log of type spec have a title related to validation checks?

Reply:
It's because there's a spec that's wrapping the validation checks, I simply done this as we're generally combining both spec + validation failures and didn't want to double count a validation failure.

<img width="780" alt="openactive-integration-tests (zsh) 2020-05-15 16-24-31" src="https://user-images.githubusercontent.com/705668/82067411-98d5de80-96c8-11ea-927e-8479c283d778.png">